src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -952,6 +956,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
```

Comment:
Just to confirm from an experiment that I did with this in Java 8 (on OSX at least), if an attempt is made to register a watcher for a regular file (as opposed to a directory), a `java.nio.file.NotDirectoryException` exception is thrown.  So we probably are stuck with needing to watch the parent directory.  Adding an option to avoid the recursive watch, though, would seem like a good thing for this use case.